### PR TITLE
[MCH] Move geometry into its own package.

### PR DESCRIPTION
**Pull request by aphecetche**
And split it into Creator, Transformer and Test parts.

This commit is massive because a lot of files were moved around, sorry
about that.

It does add *one* new feature : the ability to read geometry transformations
 from a json file, and to create such a file from a geometry.root file
 using the o2-mch-convert-geometry executable. 

For the moment the json version of the transformations is used in the unit tests.

### CODE REVIEW

**Comment by aphecetche** [comment]
@ktf @TimoWilken I'm struggling with this PR which compiles fine on my local machines (Mac and Ubuntu) but completely fails in the CI. 

The very first thing that is failing is clang-format. I thought the clang-formatting thing was kind of solved by an automatic PR on the original branch, but this is apparently failing for some reason here ?

**Comment by TimoWilken** [comment]
Hi @aphecetche, that's an interesting error. It should usually make an automatic commit, but pushing it fails for some reason I don't quite understand yet. Please could you make the following changes manually to appease `clang-format`? (Note the space after the `&`.)

```diff
diff --git a/Detectors/MUON/MCH/Simulation/src/Digitizer.cxx b/Detectors/MUON/MCH/Simulation/src/Digitizer.cxx
index 555add712..8ea0b4514 100644
--- a/Detectors/MUON/MCH/Simulation/src/Digitizer.cxx
+++ b/Detectors/MUON/MCH/Simulation/src/Digitizer.cxx
@@ -187,7 +187,7 @@ int Digitizer::processHit(const Hit& hit, int detID, int event_time)
 void Digitizer::generateNoiseDigits()
 {
 
-  o2::mch::mapping::forEachDetectionElement([&digits = this->mDigits, &normProbNoise = this->mNormProbNoise,
+  o2::mch::mapping::forEachDetectionElement([& digits = this->mDigits, &normProbNoise = this->mNormProbNoise,
                                              &eventTime = this->mEventTime, &eventID = this->mEventID,
                                              &srcID = this->mSrcID, &mcTruthOutputContainer = this->mMCTruthOutputContainer](int detID) {
     auto& seg = segmentation(detID);
@@ -213,7 +213,7 @@ void Digitizer::mergeDigits()
 {
   std::vector<int> indices(mDigits.size());
   std::iota(begin(indices), end(indices), 0);
-  std::sort(indices.begin(), indices.end(), [&digits = this->mDigits, this](int a, int b) {
+  std::sort(indices.begin(), indices.end(), [& digits = this->mDigits, this](int a, int b) {
     return (getGlobalDigit(digits[a].getDetID(), digits[a].getPadID()) < getGlobalDigit(digits[b].getDetID(), digits[b].getPadID()));
   });
```

I'm sorry, the other CI builders are still broken. Working on it!

**Comment by aphecetche** [comment]
@TimoWilken thanks for the format fix (now I have, on my side, to understand why my local clang-format is not complaining as well...). And good luck for the other CI fixes ;-)

**Comment by aphecetche** [comment]
For the record, the clang-format error is "simply" due to clang-format 10 (which is now the "default" on Mac or Ubuntu 20) behaving differently from its predecessors : 

```
~/tmp$ cat testFormat.cxx
#include <iostream>

int main()
{

  int b = 42;
  auto lam = [&a = b]() {
    return a;
  };

  return lam();
}
~/tmp$ clang-format-8 testFormat.cxx | grep auto
  auto lam = [& a = b]() {
~/tmp$ clang-format-9 testFormat.cxx | grep auto
  auto lam = [& a = b]() {
~/tmp$ clang-format-10 testFormat.cxx | grep auto
  auto lam = [&a = b]() {
```

(tests above on Ubuntu 20.04)

Now the question is whether there's an option to set in `.clang-format` to get the same behaviour in clang-format 10 for this extra space in the lamba capture part as in clang-format-8 (8 is the version that is used in the CI, right ? )

**Comment by TimoWilken** [comment]
@aphecetche yes, the CI job uses clang-format-8. Perhaps `PointerAlignment` applies here (see [docs](https://clang.llvm.org/docs/ClangFormatStyleOptions.html))? The description of `DerivePointerAlignment` seems to imply that `PointerAlignment` also applies to `&`, not just `*`.

File: Detectors/MUON/MCH/Geometry/Creator/include/MCHGeometryCreator/Geometry.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ *
+ * @param topVolume the volume the MCH geometry is attached topVolume
+ */
+void createGeometry(TGeoManager& geom, TGeoVolume& topVolume);
+
+/** get a list of MCH sensitive volumes.
+ * @returns a vector of all the MCH volumes that participate in the 
+ * particle tracking (in the transport sense).
+ */
+std::vector<TGeoVolume*> getSensitiveVolumes();
```
Comment (sawenzel) [inline Detectors/MUON/MCH/Geometry/Creator/include/MCHGeometryCreator/Geometry.h:43]: better return const& and make this function `const`

**Comment by aphecetche** [comment]
@TimoWilken seems the `PointerAlignment` is not related. At least it does not fix the issue. Anyway, let's assume it's a `clang-format` bug|feature and let's forget about it for the moment...

Concerning this PR : once the CI builders are back to their duties, will they pick it up automatically or should I force push or something ?

**Comment by TimoWilken** [comment]
@aphecetche when the CI builders are working again, I'll make sure they pick up all the broken builds. No need to push anything!

**Comment by TimoWilken** [comment]
@aphecetche the following error in build/O2/fullCI seems genuine:
```
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/Readout/v1.5.5/v1.5.5/src/ConsumerFileRecorder.cxx: In member function 'int ConsumerFileRecorder::createFile(std::shared_ptr<FileHandle>*, const DataSourceId&, bool, int)':
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/Readout/v1.5.5/v1.5.5/src/ConsumerFileRecorder.cxx:268:42: error: '%03d' directive output may be truncated writing between 3 and 10 bytes into a region of size 4 [-Werror=format-truncation=]
       snprintf(sFileId, sizeof(sFileId), "%03d", fileId);
                                          ^~~~~~
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/Readout/v1.5.5/v1.5.5/src/ConsumerFileRecorder.cxx:268:42: note: directive argument in the range [1, 2147483647]
```

**Comment by aphecetche** [comment]
@TimoWilken yes, but is not at all related to this PR, actually. It's in Readout, maybe @sy-c can comment on this one ?

**Comment by TimoWilken** [comment]
Hm, this should be fixed by AliceO2Group/Readout#151. The test failed just before alisw/alidist#2649 was merged, which uses the updated Readout. I'll mark it as pending and let it re-run with the new alidist.

**Comment by aphecetche** [comment]
@kft @TimoWilken what's the status of the Mac builders ?

**Comment by ktf** [comment]
https://alice.its.cern.ch/jira/browse/O2-1883#

**Review by sawenzel — APPROVED** [review]

**Pull request merged**